File: openapi/code_samples/csharp/api_currency_list.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;
using System.Collections.Generic;

[ApiController]
[Route("api/[controller]")]
public class CurrencyController : ControllerBase
{
    [HttpGet]
    [Route("currencies")]
    [SwaggerOperation(Summary = "Retrieves the list of supported currencies.")]
    [ProducesResponseType(typeof(List<string>), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult GetAllCurrencies()
    {
        List<string> supportedCurrencies = new List<string>
        {
            "USD",
            "EUR",
            "GBP",
            "JPY",
            // Add more currencies here
        };

        return Ok(supportedCurrencies);
    }
}

File: openapi/code_samples/python/api_currency_list.py
from flask import Flask, jsonify

app = Flask(__name__)

@app.route('/api/currency/currencies', methods=['GET'])
def get_all_currencies():
    supported_currencies = ["USD", "EUR", "GBP", "JPY"]
    # Add more currencies here
    return jsonify(supported_currencies), 200

if __name__ == '__main__':
    app.run()
